bve-native/src/lib.rs

```diff
@@ -36,3 +51,151 @@
 #![allow(clippy::shadow_same)]
 #![allow(clippy::unreachable)]
 #![allow(clippy::wildcard_enum_match_arm)]
+
+pub mod panic;
+
+use std::borrow::Cow;
+use std::ffi::{CStr, CString};
+use std::os::raw::*;
+use std::ptr::null_mut;
+
+/// C safe wrapper for an [`Option`].
+///
+/// # Safety
+///
+/// Reading from the `value` member is undefined behavior if `exists` is false. In practice it is zeroed.
```

Comment:
This is more information than I should be given

Reply:
What do you want to be given?

Reply:
You just told me that this is `sizeof(T)` bytes of memory which I do not have to allocate and may use as I please